interp/declare.ml

```diff
@@ -39,6 +39,9 @@ type section_variable_entry =
 
 type variable_declaration = DirPath.t * section_variable_entry * logical_kind
 
+let register_side_effect = ref (fun _ -> assert false)
+(** This is a forward hook to a function defined later in this file. *)
```

Comment:
why

Reply:
I thought that there was a loop in the definition, but after careful scrutiny it seems it is not the case. I'll update the PR.